src/main/java/io/fabric8/runsh/RunShLoader.java

```diff
@@ -30,36 +95,60 @@ public static String getReadme() {
         return loadFromClassPath(LOCATION_README);
     }
 
+    /**
+     * Get the run script it self
+     *
+     * @return runscript as file
+     */
+    public static String getRunScript() {
+        return loadFromClassPath(LOCATION_RUN_SCRIPT);
+    }
 
     /**
      * Copy the run scripts to a destination in the file system
      *
      * @param destination where to copy run script. Must be a directory.
+     * @return return the file to the script
      */
-    public static void copyRunScript(File destination) throws IOException {
-        if (!destination.isDirectory()) {
-            throw new IOException(String.format("Destination %s is not a directory", destination));
-        }
-        for (String script : LOCATION_RUN_SCRIPTS) {
-            File targetFile = new File(destination, FilenameUtils.getName(script));
-            try (FileWriter out = new FileWriter(targetFile)) {
-                IOUtils.copy(getInputStream(script), out);
+    public static File copyRunScript(File destination) throws IOException {
+        Path targetPath;
+        if (destination.isDirectory()) {
+            targetPath = new File(destination, RUN_SCRIPT).toPath();
+        } else {
+            if (!destination.getAbsoluteFile().getParentFile().exists()) {
+                throw new IOException(String.format("%s is not a directory", destination.getParentFile()));
             }
+            targetPath = destination.toPath();
         }
+        Files.copy(getInputStream(LOCATION_RUN_SCRIPT), targetPath, StandardCopyOption.REPLACE_EXISTING);
+        Set<PosixFilePermission> perms = new HashSet<>(Files.getPosixFilePermissions(targetPath));
+        perms.add(PosixFilePermission.OWNER_EXECUTE);
+        perms.add(PosixFilePermission.GROUP_EXECUTE);
+        perms.add(PosixFilePermission.OTHERS_EXECUTE);
+        Files.setPosixFilePermissions(targetPath, perms);
+        return targetPath.toFile();
     }
 
     // ==================================================================================
 
     private static String loadFromClassPath(String location) {
-        try {
-            return IOUtils.toString(getInputStream(location));
-        } catch (IOException e) {
-            throw new IllegalStateException("Internal: Cannot load " + location + ":" + e,e);
-        }
+        Scanner s = new Scanner(getInputStream(location)).useDelimiter("\\A");
```

Comment:
Maybe closing the input stream would be better with something like:
```
try(java.util.Scanner s = new java.util.Scanner(is)) { 
    return s.useDelimiter("\\A").hasNext() ? s.next() : ""; 
}
```

Reply:
✅